sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1011,6 +1016,139 @@ If a node containing a dynamic parameter is updated through the whole graph
 update API, then any previous updates to the dynamic parameter will be reflected
 in the new graph.
 
+==== Graph-Owned Memory Allocations [[graph-memory-allocations]]
+
+It can be desirable for a graph to own and manage memory allocations for memory
+associated with commands in the graph. This is made possible by using the
+`async_malloc` and `async_free` functions from the
+link:../experimental/sycl_ext_codeplay_async_memory_alloc.asciidoc[sycl_ext_codeplay_async_memory_alloc
+] extension. These commands can be added to a graph either via queue recording
+or explicit graph creation, which will create allocations which are owned and
+managed by that specific `command_graph`.
+
+This can be useful for example when capturing library calls, as the library may
+require some temporary memory for commands they record to the graph. Use of
+these APIs by the library enables the creation and management of these
+allocations within the graph, without the library needing to manage their
+lifetime or be aware that the commands are being recorded to a graph.
+
+Graphs use vitual memory to manage these allocations. To use this feature in a
+graph the device associated with the graph must support the
+`ext_oneapi_virtual_mem` aspect. Attempting to use this feature with devices
+that do not support this aspect will result in synchronous errors being thrown.
+
+Pointers returned from allocation nodes can be used in other graph nodes in the
+same way as regular USM pointers.
+
+===== Supported Features
+
+Currently only device allocations are supported in graphs. Attempting to
+add allocations of any other type to a graph will result in synchronous errors
+being thrown.
+
+===== Restrictions
+
+The following restrictions apply to any graph using these allocations:
+
+* The graph cannot be finalized more than once.
+* The graph cannot be used as a sub-graph in another graph.
+* Graph memory allocation nodes cannot be updated, and graphs containing these
+nodes cannot be updated via <<whole-graph-update, Whole Graph Update>>.
+
+===== Behaviour 
+
+The semantics of `async_malloc` and `async_free` within a graph differ from the
+eager SYCL usage described in the
+link:../experimental/sycl_ext_codeplay_async_memory_alloc.asciidoc[sycl_ext_codeplay_async_memory_alloc]
+extension.
+
+* Graph memory allocations are not made directly from any default or
+user-provided Memory Pool. Each graph containing alloc/free nodes maintains its
+own pool of memory from which allocations are made. The following properties
+of a default or user-provided memory pool provided in calls to `async_malloc`
+will be respected for the associated graph allocations:
```

Comment:
should way say explicitly that any other properties will be ignored?

Reply:
Good point, I've done this and also added that it will respect the allocation kind of the pool as that was missing before.